packages/core/admin/shared/contracts/api-token.ts

```diff
@@ -1,9 +1,26 @@
 import { errors } from '@strapi/utils';
 
-import type { ApiToken } from '../../server/src/services/api-token';
+import { ValidationError as YupValidationError } from 'yup';
+import { Entity } from '@strapi/types';
 
-type ApiTokenBody = Pick<ApiToken, 'lifespan' | 'description' | 'type' | 'name' | 'permissions'>;
-type ApiTokenResponse = Omit<ApiToken, 'accessKey'>;
+type ApiToken = {
+  accessKey: string;
+  createdAt: string;
+  description: string;
+  expiresAt: string;
+  id: Entity.ID;
+  lastUsedAt: string | null;
+  lifespan: string;
+  name: string;
+  permissions: any[];
```

Comment:
is this not `Permission[]`? i.e. an actual permission shape

Reply:
I think it exists, but it was very impractical on the FE because inputs etc. don't use generics so they handle `string`.... maybe just do `string[]` because that also means it might be empty.